### PR TITLE
Allowing message transformation before producing

### PR DESCRIPTION
**Pull request by gustavo-momente**

### CODE REVIEW

File: kafka_topic_dumper/kafka_client.py [hunk 1]
```diff
@@ -378,17 +379,19 @@ def _load_dump(self, bucket_name, dump_id, download_dir, files):
...
-                for row in df.itertuples():
-                    future = self.producer.send(self.topic, key=row[1],
-                                                value=row[2])
-                    future.get(timeout=self.timeout)
+                for raw_row in df.itertuples():
+                    for row in transformer_class.transform(raw_row):
+                        self.producer.send(self.topic, key=row[1],
+                                           value=row[2])
                 logger.debug('File <{}> reloaded to kafka'.format(file_path))
+                self.producer.flush()
```
Comment (Evnsan) [inline kafka_topic_dumper/kafka_client.py]: Could we use `self.producer.flush(timeout=self.timeout)` here?

File: kafka_topic_dumper/main.py [hunk 2]
```diff
@@ -114,7 +130,12 @@ def main():
                 if not dump_id:
                     dump_id = kafka_client.find_latest_dump_id(bucket_name)
                     logger.info('Using latest dump id <{}>'.format(dump_id))
+                cl = get_transformer_class(transformer)
```
Comment (Evnsan) [inline kafka_topic_dumper/main.py]: We should use more descriptive variable names. Something like `transformer_class = get_transformer_class(transformer)`

File: kafka_topic_dumper/main.py [hunk 3]
```diff
@@ -66,6 +66,11 @@ def parse_command_line():
                                  'will not load it again, it will only reset '
                                  'offsets for this consumer-group.')
 
+    reload_cmd.add_argument('-T', '--transformer',
```
Comment (Evnsan) [inline kafka_topic_dumper/main.py:69]: This option creates a problem in hot-reload feature. As you can change the transformer class for the same dump file we need to create a force reload option.

File: kafka_topic_dumper/main.py [hunk 4]
```diff
@@ -79,6 +84,16 @@ def parse_command_line():
     return opts
 
 
+def get_transformer_class(pre_processing):
+    [module_name, class_name] = pre_processing.split(":")
+
+    logger.info(module_name)
```
Comment (gustavo-momente) [inline kafka_topic_dumper/main.py]: fix me

File: kafka_topic_dumper/main.py [hunk 5]
```diff
@@ -114,7 +120,10 @@ def main():
                 if not dump_id:
                     dump_id = kafka_client.find_latest_dump_id(bucket_name)
                     logger.info('Using latest dump id <{}>'.format(dump_id))
+
+
```
Comment (Evnsan) [inline kafka_topic_dumper/main.py]: extra line

**Review by Evnsan — APPROVED** [review]

**Pull request merged**